feature/experimental/bgp/ate_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go

```diff
@@ -0,0 +1,445 @@
+func bgpClearConfig(t *testing.T, dut *ondatra.DUTDevice) {
+	resetBatch := &gnmi.SetBatch{}
+	gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Config())
+
+	if deviations.NetworkInstanceTableDeletionRequired(dut) {
+		tablePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).TableAny()
+		for _, table := range gnmi.LookupAll(t, dut, tablePath.Config()) {
+			if val, ok := table.Val(); ok {
+				if val.GetProtocol() == oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP {
+					gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Table(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, val.GetAddressFamily()).Config())
+				}
+			}
+		}
+	}
+	resetBatch.Set(t, dut)
+}
+
+func verifyPeer(t *testing.T, nbr *bgpNbr, dut *ondatra.DUTDevice) {
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(nbr.peerIP)
+	glblPath := statePath.Global()
+
+	// Check BGP peerAS from telemetry.
+	peerAS := gnmi.Get(t, dut, nbrPath.State()).GetPeerAs()
+	if peerAS != nbr.peerAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, nbr.peerAS)
+	}
+
+	// Check BGP localAS from telemetry.
+	localAS := gnmi.Get(t, dut, nbrPath.State()).GetLocalAs()
+	if localAS != nbr.localAS {
+		t.Errorf("BGP localAS: got %v, want %v", localAS, nbr.localAS)
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != nbr.localAS {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+
+	verifyPrefixesTelemetry(t, dut, 3, nbr.isV4)
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string, prefixes []string) *ondatra.ATETopology {
+	t.Helper()
+	t.Log("Configure ATE interface")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+	iDut1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
+
+	bgpDut1 := iDut1.BGP()
+	peer := bgpDut1.AddPeer().WithPeerAddress(ateParams.peerIP).WithLocalASN(ateParams.localAS)
+
+	if connectionType == connInternal {
+		peer.WithTypeInternal()
+	} else {
+		peer.WithTypeExternal()
+	}
+
+	if ateParams.isV4 {
+		network := iDut1.AddNetwork("bgpNeti1")
+		network.IPv4().WithAddress(prefixes[0]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 4400, 3300)
+
+		network = iDut1.AddNetwork("bgpNeti2")
+		network.IPv4().WithAddress(prefixes[1]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 7700)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4).Communities().WithPrivateCommunities("200:1")
```

Comment:
no need to set NextHopAddress here; it is already set on the previous line

Reply:
Addressed with latest commit